src/api/handlers/deviceprofile.py

```diff
@@ -113,6 +115,75 @@ def log_device(self, request):
     if err:
       return MassenergizeResponse(error=str(err), status=err.status)
     return MassenergizeResponse(data=device)
+
+  @super_admins_only
+  def platform_metrics(self, request):
+    context: Context = request.context
+    args: dict = context.args
+
+    self.validator.expect("metric", str, is_required=True)
+    self.validator.expect("community_id", str)
+    args, err = self.validator.verify(args)
+
+    if err:
+      return err
+
+    metric = args["metric"] if "metric" in args else None
+    community_id = args["community_id"] if "community_id" in args else None
+    
+    if metric is "anonymous_users":
+      metric, err = self.service.metric_anonymous_users(context, args)
+      if err:
+        return MassenergizeResponse(error=str(err), status=err.status)
+
+    if metric is "anonymous_community_users":
+      metric, err = self.service.metric_anonymous_community_users(context, args, community_id)
+      if err:
+        return MassenergizeResponse(error=str(err), status=err.status)
+
+    if metric is "user_profiles":
+      metric, err = self.service.metric_user_profiles(context, args)
+      if err:
+        return MassenergizeResponse(error=str(err), status=err.status)
+
+    if metric is "community_profiles":
+      metric, err = self.service.metric_community_profiles(context, args, community_id)
+      if err:
+        return MassenergizeResponse(error=str(err), status=err.status)
+        
+    return MassenergizeResponse(data=metric)
+  
+  @admins_only
+  def community_metrics(self, request):
+    context: Context = request.context
+    args: dict = context.args
+
+    self.validator.expect("metric", str, is_required=True)
+    self.validator.expect("community_id", str)
```

Comment:
This route looks like it would pass junk back if community_id not specified.  Should it be required?

Reply:
and if the metric is something different than the three choices, should it return an error?